### PR TITLE
support pasting multiple magnet links

### PR DESCRIPTION
**Pull request by Coeur**
We already support copying multiple magnet links.
This commit adds the symmetric support to paste multiple magnet links from a single string.

### CODE REVIEW

**Comment by ckerr** [comment]
Looks OK to me?

Ping @DevilDimon

File: macosx/TorrentTableView.mm [hunk 1]
```diff
@@ -667,11 +667,16 @@ - (void)paste:(id)sender
...
         {
-            NSDataDetector* detector = [NSDataDetector dataDetectorWithTypes:NSTextCheckingTypeLink error:nil];
-            for (__strong NSString* pbItem in items)
+            return;
+        }
+        NSDataDetector* detector = [NSDataDetector dataDetectorWithTypes:NSTextCheckingTypeLink error:nil];
+        for (NSString* itemString in items)
+        {
+            NSArray<NSString*>* itemLines = [itemString componentsSeparatedByCharactersInSet:NSCharacterSet.newlineCharacterSet];
+            for (__strong NSString* pbItem in itemLines)
```
Comment (DevilDimon) [inline macosx/TorrentTableView.mm:679]: `__strong` is not needed as the reference is strong by default

**Review by DevilDimon — APPROVED** [review]

File: macosx/TorrentTableView.mm (hunk 1 above)
Comment (Coeur) [reply macosx/TorrentTableView.mm:679]: It is needed explicitly in ARC's fast enumeration when the object is re-assigned, which is the case two lines below.
Try to remove `__strong` and it won't compile anymore.

Comment (DevilDimon) [reply macosx/TorrentTableView.mm:679]: Right, I thought it was an artefact from the MRR-ARC conversion of years ago. LGTM

**Pull request merged**